scanpy/tests/notebooks/test_pbmc3k.py

```diff
@@ -115,32 +116,32 @@ def test_pbmc3k(image_comparer):
 
     sc.tl.rank_genes_groups(adata, "leiden")
     sc.pl.rank_genes_groups(adata, n_genes=20, sharey=False, show=False)
-    save_and_compare_images("rank_genes_groups_1")
+    save_and_compare_images_rank_genes("rank_genes_groups_1")
 
     sc.tl.rank_genes_groups(adata, "leiden", method="logreg")
     sc.pl.rank_genes_groups(adata, n_genes=20, sharey=False, show=False)
-    save_and_compare_images("rank_genes_groups_2")
+    save_and_compare_images_rank_genes("rank_genes_groups_2")
 
     sc.tl.rank_genes_groups(adata, "leiden", groups=["0"], reference="1")
     sc.pl.rank_genes_groups(adata, groups="0", n_genes=20, show=False)
-    save_and_compare_images("rank_genes_groups_3")
+    save_and_compare_images_rank_genes("rank_genes_groups_3")
 
     # gives a strange error, probably due to jitter or something
     # sc.pl.rank_genes_groups_violin(adata, groups='0', n_genes=8)
     # save_and_compare_images('rank_genes_groups_4')
 
-    if adata[adata.obs["leiden"] == "4", "CST3"].X.mean() < 1:
+    if adata[adata.obs["leiden"] == "3", "CST3"].X.mean() < 1:
         (  # switch clusters
+            adata.obs["leiden"][adata.obs["leiden"] == "3"],
             adata.obs["leiden"][adata.obs["leiden"] == "4"],
-            adata.obs["leiden"][adata.obs["leiden"] == "5"],
-        ) = ("5", "4")
+        ) = ("4", "3")
```

Comment:
The 4-5 swap is now a 3-4 swap

Reply:
![image](https://github.com/scverse/scanpy/assets/43999641/e6f2b563-7725-4820-b775-a9ef541dd071)

![image](https://github.com/scverse/scanpy/assets/43999641/f07b6aec-ab07-4ec8-b096-2bb2324ebfe9)